src/zodotenv.spec.ts

```diff
@@ -102,4 +102,98 @@ describe('zodotenv', () => {
       assert.deepEqual(config('adminCredentials'), { name: 'admin', password: '12345' });
     });
   });
+
+  describe('Serialization', () => {
+    const originalEnv = structuredClone(process.env);
+
+    afterEach(() => {
+      process.env = originalEnv;
+    });
+
+    it('serializes the entire configuration object to JSON', () => {
+      process.env.PORT = '3000';
+      process.env.HTTP2 = 'false';
+      process.env.DB_HOST = 'localhost:5432';
+      process.env.DB_DRIVER = 'mysql';
+      process.env.DB_TABLES = 'users,posts';
+      process.env.ADMIN_CREDENTIALS = '{"name": "admin", "password": "secret"}';
+
+      const config = zodotenv({
+        name: ['NAME', z.string().default('my-app')],
+        port: ['PORT', z.coerce.number()],
+        http2: ['HTTP2', z.string().transform((s) => s === 'true')],
+        database: {
+          host: ['DB_HOST', z.string()],
+          driver: ['DB_DRIVER', z.enum(['mysql', 'pgsql', 'sqlite'])],
+          tables: ['DB_TABLES', z.preprocess((s) => String(s).split(','), z.array(z.string()))],
+        },
+        adminCredentials: [
+          'ADMIN_CREDENTIALS',
+          z
+            .string()
+            .transform((s) => JSON.parse(s))
+            .pipe(
+              z.object({
+                name: z.string(),
+                password: z.string(),
+              }),
+            ),
+        ],
+      });
+
+      const expectedConfig = {
+        name: 'my-app',
+        port: 3000,
+        http2: false,
+        database: {
+          host: 'localhost:5432',
+          driver: 'mysql',
+          tables: ['users', 'posts'],
+        },
+        adminCredentials: {
+          name: 'admin',
+          password: 'secret',
+        },
+      };
+
+      assert.deepEqual(config.toJSON(), expectedConfig);
```

Comment:
- is `toJSON` the best name here? `expectedConfig` is an object, not a JSON string, but these are equal here
- personally I would prefer the see a flat output (where the keys are the object-path-style strings that you pass into `config()`) rather than the nested output which mirrors the configuration object. if you insist on having the nested output then this could be a param of thing function like `.toJSON({ nested: true })`

Reply:
* I was wondering what the best name here would be too - had these options in mind: `toObject`, `asObject`, `export` and `dump` - happy to hear more 🙂
* do you mean that with config of: 
```ts
const config = zodotenv({
  port: ['PORT', z.coerce.number()],
  secret: ['SECRET', z.string()]
})
```
you would get the following output: 
```json
{
  "PORT": "3000",
  "SECRET": "foo"
}
```
I like this better tbh ^ - happy to change it to that

Reply:
I'm happy to keep `.toJSON()` as that is the de-facto standard in JS to format an object for serialisation (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description), but first I founded it weird, but I'm okay with that now after thinking about it for a few mins.

By "flat" I mean this:
```json
{
  "port": 3000,
  "database.host": "localhost:5432",
  "database.drive": "mysql",
  "database.tables": ["users", "posts"]
}
```

So basically a dump of the internal Map in `config`.

Reply:
gotcha - yep that sounds good!